### PR TITLE
Docs

### LINKED ISSUE #122: API documentation
API documentation is needed. Check https://www.gtk.org/gtk-doc/ and http://www.doxygen.nl/.

>  Doxygen is the de facto standard tool for generating documentation from annotated C++ sources

### PR DESCRIPTION
**Pull request by azubieta**
Generate docs using doxygen and exhale.

Travis is failing to deploy to github pages. Don't know if it's because I don't have enough permissions @TheAssassin would you mind to take a look at it?

fixes #122

### CODE REVIEW

**Comment by TheAssassin** [comment]
Why bother deploying this on GitHub pages?

**Comment by azubieta** [comment]
> Why bother deploying this on GitHub pages?

To make it available online without requiring an special hosting? If you have an idea of where to deploy it please let me know

**Comment by TheAssassin** [comment]
I thought we might integrate it as part of the AppImage docs. Or at least serve it under docs.appimage.org

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'LibAppImage'
```
Comment (TheAssassin) [inline docs/conf.py]: Lower case for library names. It's libappimage.

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'breathe',
+    'exhale'
+]
+
+# Setup the breathe extension
+breathe_projects = {
+    "LibAppImage": "./doxyoutput/xml"
```
Comment (TheAssassin) [inline docs/conf.py]: Should be "libappimage", too.

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+# Setup the exhale extension
+exhale_args = {
+    # These arguments are required
+    "containmentFolder": "./api",
+    "rootFileName": "library_root.rst",
+    "rootFileTitle": "LibAppImage API",
```
Comment (TheAssassin) [inline docs/conf.py]: Lower case "libappimage".

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+
+# Tell sphinx what the primary language being documented is.
+primary_domain = 'cpp'
+
+# Tell sphinx what the pygments highlight language should be.
+highlight_language = 'cpp'
+
+breathe_default_project = "LibAppImage"
```
Comment (TheAssassin) [inline docs/conf.py]: "libappimage"

**Comment by probonopd** [comment]
Where can one see the docs?

**Comment by probonopd** [comment]
Anyone?

**Pull request merged**